api/datastore/sql/sql.go

```diff
@@ -508,11 +509,16 @@ func (ds *SQLStore) GetAppByID(ctx context.Context, appID string) (*models.App,
 }
 
 // GetApps retrieves an array of apps according to a specific filter.
-func (ds *SQLStore) GetApps(ctx context.Context, filter *models.AppFilter) ([]*models.App, error) {
-	res := []*models.App{} // for JSON empty list
+func (ds *SQLStore) GetApps(ctx context.Context, filter *models.AppFilter) (*models.AppList, error) {
+	res := &models.AppList{}
 
-	if filter.NameIn != nil && len(filter.NameIn) == 0 { // this basically makes sure it doesn't return ALL apps
-		return res, nil
+	if filter.Cursor != "" {
+		s, err := base64.RawURLEncoding.DecodeString(filter.Cursor)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Error(s)
+		filter.Cursor = string(s)
```

Comment:
as  above - feels like the decoding should be in buildFiterAppQuery..

Reply:
it should, but that doesn't have an error return...